miner/src/worker/mod.rs

```diff
@@ -0,0 +1,112 @@
+mod cuckoo_simple;
+mod dummy;
+
+use crate::config::{WorkerConfig, WorkerType};
+use ckb_core::header::Seal;
+use ckb_logger::error;
+use ckb_pow::{CuckooEngine, DummyPowEngine, PowEngine};
+use crossbeam_channel::{unbounded, Sender};
+use cuckoo_simple::CuckooSimple;
+use dummy::Dummy;
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use numext_fixed_hash::H256;
+use std::sync::Arc;
+use std::thread;
+
+#[derive(Clone)]
+pub enum WorkerMessage {
+    Stop,
+    Start,
+    NewWork(H256),
+}
+
+pub struct WorkerController {
+    inner: Vec<Sender<WorkerMessage>>,
+}
+
+impl WorkerController {
+    pub fn new(inner: Vec<Sender<WorkerMessage>>) -> Self {
+        Self { inner }
+    }
+
+    pub fn send_message(&self, message: WorkerMessage) {
+        for worker_tx in self.inner.iter() {
+            if let Err(err) = worker_tx.send(message.clone()) {
+                error!("worker_tx send error {:?}", err);
+            };
+        }
+    }
+}
+
+pub fn start_worker(
+    pow: Arc<dyn PowEngine>,
+    config: &WorkerConfig,
+    seal_tx: Sender<(H256, Seal)>,
+) -> WorkerController {
+    let mp = MultiProgress::new();
+    let spinner_style = ProgressStyle::default_spinner()
```

Comment:
How it resolves the conflicts with logger?

Reply:
progress bar log to stderr, no conflict with the logger.

Reply:
Some may want to run miner as a daemon. The progress should be disabled if so. Let's add `StandardError=null` under `[Service]` in devtools/init/linux-systemd/ckb-miner.service